### PR TITLE
Add `system sync filesystem cache` command

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `SYSTEM SYNC FILESYSTEM CACHE` command. It will compare in-memory state of filesystem cache with what it has on disk and fix in-memory state if needed.

### CODE REVIEW

File: src/Interpreters/Cache/FileCache.h [hunk 1]
```diff
@@ -102,11 +102,11 @@ class FileCache : private boost::noncopyable
...
     bool tryReserve(FileSegment & file_segment, size_t size);
 
-    FileSegmentsHolderPtr getSnapshot();
+    FileSegments getSnapshot();
 
-    FileSegmentsHolderPtr getSnapshot(const Key & key);
+    FileSegments getSnapshot(const Key & key);
 
-    FileSegmentsHolderPtr dumpQueue();
+    FileSegments dumpQueue();
```
Comment (serxa) [inline src/Interpreters/Cache/FileCache.h:109]: Is it okay to make a copy of the entire list? At first glance, seems expensive.

Comment (kssenii) [reply src/Interpreters/Cache/FileCache.h:109]: `dumpQueue` is used for unit test only, and `getSnapthot` is only for system tables. Also afaik the the result is moved and not copied as the function returns a local variable.

**Review by serxa — APPROVED** [review]

**Comment by kssenii** [comment]
The tests in this pr fail because of revert of https://github.com/ClickHouse/ClickHouse/pull/52778.
So waiting for it to be merged and then this PR will be ok.

**Comment by kssenii** [comment]
> Stress test (asan) — Killed by signal

`Logical error: 'Cannot find transaction (115, 33, 5a0ac955-d43c-4a5d-8392-a3aca7bdba75) that has started mutation mutation_12.txt that is going to be applied to part all_7_7_0'`
Issue: https://github.com/ClickHouse/ClickHouse/issues/53615

**Comment by kssenii** [comment]
> Integration tests (release) [2/4]

test_storage_rabbitmq

**Pull request merged**